File: employeeManager/src/main/java/com/infy/employeeManager/service/EmployeeService.java
package com.infy.employeeManager.service;

import java.util.List;

import org.springframework.stereotype.Service;

import com.infy.employeeManager.dto.EmployeeDTO;
import com.infy.employeeManager.exception.EmployeeManagerException;

@Service
public interface EmployeeService {
	public EmployeeDTO getEmployeeById(long empoyeeId) throws EmployeeManagerException;
	public List<EmployeeDTO> getAllEmployees() throws EmployeeManagerException;
	public Long addEmployee(EmployeeDTO employeeDTO) throws EmployeeManagerException;
	public void removeEmployee(long employeeId) throws EmployeeManagerException;
	public List<EmployeeDTO> findEmployeesByPincode(long pincode) throws EmployeeManagerException;
	public EmployeeDTO updateEmployee(long emplpoyeeId, EmployeeDTO empDto) throws EmployeeManagerException;
	public List<EmployeeDTO> getAllEmployees(int pageNo, int pageSize) throws EmployeeManagerException;
}

File: empman-ui/src/services/EmployeeService.js
import axios from 'axios'

const EMPLOYEE_BASE_REST_API_URL = 'http://localhost:8080/employee';

class EmployeeService{

    getAllEmployees(){
        return axios.get(EMPLOYEE_BASE_REST_API_URL+"/getAllEmployees")
    }

    getEmployeesByPage(pageNo, pageSize){
        return axios.get(EMPLOYEE_BASE_REST_API_URL+"/pagination/"+ pageNo + "/" + pageSize)
    }

    createEmployee(employee){
        return axios.post(EMPLOYEE_BASE_REST_API_URL+"/addemployee", employee)
    }

    getEmployeeById(employeeId){
        return axios.get(EMPLOYEE_BASE_REST_API_URL + '/getEmployee/' + employeeId);
    }

    updateEmployee(employeeId, employee){
        return axios.put(EMPLOYEE_BASE_REST_API_URL + '/update/' +employeeId, employee);
    }

    deleteEmployee(employeeId){
        return axios.delete(EMPLOYEE_BASE_REST_API_URL + '/deleteEmployee/' + employeeId);
    }
}

export default new EmployeeService();